### ISSUE TITLE
Custom report error: dynamic filters for select fields

### ISSUE THREAD

**Issue by sevendaysdigital** [mautic/mautic#7075]
What type of report is this:

| Q  | A
| ---| ---
| Bug report? | V
| Feature request? | 
| Enhancement? | 

## Description:
When adding a dynamic filter to your report and in case the filter field is a select list, the filter values are not visible in the report:

eg I've defined a custom select field "Type" with 2 values: 
![screen shot 2019-01-02 at 13 06 50](https://user-images.githubusercontent.com/20223463/50591329-5fb01c00-0e8f-11e9-9903-a0d1eefe639a.png)

In the reports section, I've added this field as dynamic filter:

![screen shot 2019-01-02 at 13 08 45](https://user-images.githubusercontent.com/20223463/50591367-8f5f2400-0e8f-11e9-956c-dcc27acbc26f.png)

When trying to change the filter value in the report, no values are found for the field:

![screen shot 2019-01-02 at 13 09 43](https://user-images.githubusercontent.com/20223463/50591401-b9184b00-0e8f-11e9-9e2c-f0bb19704ea9.png)

Hope anyone can help on this one?

## If a bug:

| Q   | A
| --- | ---
| Mautic version | 2.14.2 + 2.15.0
| PHP version | 7.1.23

### Steps to reproduce:
1. Edit a report
2. Add a custom dynamic filter field based on a select field
3. View the report and try to filter on the dynamic field, no values are found for the filter field.

**Comment by RCheesley** [comment]
I think @sevendaysdigital has a similar issue that I've found with dynamic filters while I was testing a PR.

I made a short screen capture to explain the issue I came across here: https://youtu.be/fvvoIEoVHOg.

In my case, it's confusing both to not see the condition for a dynamic filter, but also to not show the filter that is being applied by default. Instead there is simply a blank box with the field name above it.

I think this may relate to the issue above if the field/value isn't being displayed or made available?
